From da71d74a03fa8d8150c451c7a085c09f045fba49 Mon Sep 17 00:00:00 2001
From: Sebastian Lackner <sebastian@fds-team.de>
Date: Sat, 28 Feb 2015 21:52:40 +0100
Subject: winex11: Initialize thread data in clip_cursor_notify.

Fixes a crash with Slender - The Arrival when trying to start the game.
For some reason the thread local data is not initialized yet when we enter
the function.
---
 dlls/winex11.drv/mouse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/winex11.drv/mouse.c b/dlls/winex11.drv/mouse.c
index 5f5bdeb..95efa19 100644
--- a/dlls/winex11.drv/mouse.c
+++ b/dlls/winex11.drv/mouse.c
@@ -453,7 +453,7 @@ void reset_clipping_window(void)
  */
 LRESULT clip_cursor_notify( HWND hwnd, HWND new_clip_hwnd )
 {
-    struct x11drv_thread_data *data = x11drv_thread_data();
+    struct x11drv_thread_data *data = x11drv_init_thread_data();
 
     if (hwnd == GetDesktopWindow())  /* change the clip window stored in the desktop process */
     {
-- 
2.3.0

